js/src/frameworks/openai.ts

```diff
@@ -132,18 +132,22 @@ export class OpenAIToolSet extends BaseComposioToolSet {
                 runId = id;
             }
 
-            if(!runId) {
+            if (!runId) {
                 continue;
             }
 
             // Handle the 'requires_action' event
             if (event.event === 'thread.run.requires_action') {
                 const toolOutputs = await this.handleAssistantMessage(event.data, entityId);
 
-                // Submit the tool outputs
-                await client.beta.threads.runs.submitToolOutputs(thread.id, runId, {
-                    tool_outputs: toolOutputs
-                });
+                // Check if the run is still in a state that accepts tool outputs
+                const currentRun = await client.beta.threads.runs.retrieve(thread.id, runId);
```

Comment:
Consider extracting the state array into a constant at the class level to improve maintainability and prevent duplication:
```typescript
private static readonly VALID_TOOL_OUTPUT_STATES = ["queued", "in_progress", "requires_action"] as const;
```
This would also enable TypeScript to infer the literal types.